modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/Walker.java

```diff
@@ -799,7 +799,19 @@ public ANode visitNull(NullContext ctx) {
 
     @Override
     public ANode visitString(StringContext ctx) {
-        String string = ctx.STRING().getText().substring(1, ctx.STRING().getText().length() - 1);
+        String string = ctx.STRING().getText();
+        boolean doubleQuoted = string.charAt(0) == '"';
+
+        // Strip the leading and trailing quotes
+        string = string.substring(1, ctx.STRING().getText().length() - 1);
```

Comment:
We can. I figured the replacements would mostly end up as noops so it wasn't worth it. Especially with this on the compile side. If you'd prefer it I'm happy to do it though.